src/wallet/test/availablecoins_tests.cpp

```diff
@@ -55,17 +55,30 @@ class AvailableCoinsTestingSetup : public TestChain100Setup
     std::unique_ptr<CWallet> wallet;
 };
 
+void TestCoinsResult(AvailableCoinsTestingSetup& context, OutputType out_type, CAmount amount,
+                     std::map<OutputType, size_t>& expected_coins_sizes)
+{
+    LOCK(context.wallet->cs_wallet);
+    util::Result<CTxDestination> dest = Assert(context.wallet->GetNewDestination(out_type, ""));
+    CWalletTx& wtx = context.AddTx(CRecipient{{GetScriptForDestination(*dest)}, amount, /*fSubtractFeeFromAmount=*/true});
+    CoinFilterParams filter;
+    filter.skip_locked = false;
+    CoinsResult available_coins = AvailableCoins(*context.wallet, nullptr, std::nullopt, filter);
+    // Lock outputs so they are not spent in follow-up transactions
+    for (uint32_t i = 0; i < wtx.tx->vout.size(); i++) context.wallet->LockCoin({wtx.GetHash(), i});
+    for (const auto& it : available_coins.coins) BOOST_CHECK_EQUAL(it.second.size(), expected_coins_sizes[it.first]);
```

Comment:
As shortly discussed via IRC, the test currently also succeeds if `filter.skip_locked` is set to true. That is because we only iterate output types for which coins exist in `available_coins`. Should be fixed by iterating over all output types here.

Reply:
yes, good catch!